### ISSUE TITLE
embargo is applied before the validation block, where the term means after it

### ISSUE THREAD

**Issue by nbx-liz** [nbx-liz/LizyML#273]
## Summary

`purged_time_series` exposes `purge_gap` and `embargo` as two knobs. Measured
against the shipped splitter, they are one knob: both widen the same dead zone
between the end of training and the start of validation, and `embargo` never
does the thing its name denotes.

This is **not a spec/implementation drift.** `BLUEPRINT.md:587` already defines
`embargo` (the source line is Japanese; paraphrased) as *an additional number of
observations excluded between train and valid, in the same unit as
`gap` / `purge_gap`* — and the code does exactly that. The divergence is from
the term's established meaning in the literature the mechanism comes from, and
it has a measurable cost on the inner split.

Filed separately from #265, which found the specification's own inconsistency
about how that gap propagates. This issue is one level up: whether the amount
being propagated is composed correctly.

## What `embargo` means where the mechanism comes from

Purging and embargo are directional, and the direction is the whole distinction.
`skfolio`, which implements the scheme, states both:

- `purged_size` — "Number of observations to exclude from the start of each
  train set that are **after** a test set **and** the number of observations to
  exclude from the end of each training set that are **before** a test set."
- `embargo_size` — "Number of observations to exclude from the start of each
  training set that are **after** a test set."

The asymmetry has a reason. Purging answers forward-looking **labels**: a
training row whose label window reaches into the test period leaks. Embargo
answers backward-looking **features**: a training row positioned *after* the test
block may carry features computed over a window overlapping it, and purging
cannot see that. Embargo therefore needs training data after the test block.

## Evidence (HEAD `5712f41`)

`lizyml/splitters/purged_time_series.py` builds expanding-window folds: training
is `indices[:train_end]` and validation is the next chunk, so training is always
entirely before validation. `docs/config-reference.md:106` states this as a
design property — "Build each fold in chronological order (`train` always before
`valid`)".

Both knobs are subtracted at the same point:

```python
train_end = (k + 1) * fold_size - self.purge_gap - self.embargo
```

Executed over every fold:

```
PurgedTimeSeriesSplitter(n_splits=4, purge_gap=5, embargo=2), n=240

  fold 0: train=[  0.. 40] valid=[ 48.. 95]  dead zone=7  train rows AFTER valid=0
  fold 1: train=[  0.. 88] valid=[ 96..143]  dead zone=7  train rows AFTER valid=0
  fold 2: train=[  0..136] valid=[144..191]  dead zone=7  train rows AFTER valid=0
  fold 3: train=[  0..184] valid=[192..239]  dead zone=7  train rows AFTER valid=0

each knob, isolated (fold 0 dead zone):
  purge_gap=0 embargo=0 -> 0      purge_gap=0 embargo=2 -> 2
  purge_gap=5 embargo=0 -> 5      purge_gap=5 embargo=2 -> 7
```

**No fold places training data after the validation block**, so the mechanism
`embargo` names has nothing to act on. And each knob moves the same number.

## Impact

**A user who knows the term gets the opposite of what they configure.** Setting
`embargo` to guard against post-validation serial correlation instead removes
additional *pre-validation* training rows. Nothing warns, and the two knobs are
documented side by side in `docs/config-reference.md:115` under separate column
headings ("Purge" / "Embargo"), which presents them as distinct mechanisms.

**It costs training rows on the inner split, where they are scarcest.** H-0085
propagates `purge_gap + embargo` to the early-stopping split
(`BLUEPRINT.md:602`). The leakage that propagation exists to prevent is the
label-horizon overlap between inner-train and inner-valid — which `purge_gap`
alone bounds. The `embargo` term is additional purging with no leakage
justification:

```
inner split on a 192-row outer train fold, ratio=0.1
  gap=purge_gap         = 5   inner-train 168 rows,  5 purged
  gap=purge_gap+embargo = 7   inner-train 166 rows,  7 purged
```

It also tightens the bound at which `TimeHoldoutInnerValid` raises
`ValueError("Inner validation would consume all N samples")`, so a small outer
fold fails earlier than the leakage constraint requires.

**This is not a leak.** Over-purging is conservative in the safe direction. The
defect is a term of art that means something else, plus training data spent for
no stated reason.

## Defect class

Closest to **DC6 (dead-in-practice)**, applied to the mechanism the *name*
promises: post-test embargo has an activation condition — training data after the
test block — that the production geometry never satisfies, 0 of every fold.
The operational check separates it cleanly: removing post-test embargo behaviour
changes nothing, because it was never reachable. What ships under the name is
live, but it is a second purge.

Recording it this way rather than forcing a fit: the implemented behaviour is
not dead, and the classification applies to the promise, not the code.

## Suggested direction

Three options; the first is recommended.

1. **Rename to what it does.** A second gap parameter with a name that does not
   claim a directional mechanism — and a `BLUEPRINT.md` §10.2 statement that
   this splitter is forward-chaining and therefore has no position for an
   embargo. Requires a deprecation path, since `embargo` is a public Config key
   (`config/schema.py:135`) and already carries one legacy alias (`embargo_pct`,
   `BLUEPRINT.md:322`).
2. **Implement embargo in its real direction.** This needs folds whose test
   block has training data after it — an interior-test-block splitter, which
   this one does not produce. That is a new splitter, not a change to this one.
3. **Document the divergence deliberately.** Cheapest, and keeps a term that
   will mislead every reader who knows it.

Whichever is chosen, `BLUEPRINT.md:602`'s propagation rule should state the
concept ("the outer split's boundary gap") rather than the composition
`purge_gap + embargo`, so the specification does not pin an amount this issue
puts in question. #265 covers the rewrite of the surrounding clause.

## Relevant SKILL

`skills/splitters-and-splitplan/SKILL.md`,
`skills/training-cv-and-inner-valid/SKILL.md`,
`skills/config-and-specs/SKILL.md`

## DoD

- `purged_time_series` has no knob whose name asserts a mechanism the splitter's
  geometry cannot host, or the divergence is stated in `BLUEPRINT.md` §10.2 as a
  deliberate decision with its reason.
- A test asserts the geometric fact this issue rests on: over every fold of
  `PurgedTimeSeriesSplitter`, no training index exceeds the maximum validation
  index. That is the condition under which a post-test embargo could exist, and
  it should fail loudly if a future splitter change makes it true.
- The inner-split gap and the outer dead zone are derived from one stated rule,
  so the two cannot drift apart again.
- Any Config key change routes through `docs/DEPRECATIONS.md` per H-0076.
